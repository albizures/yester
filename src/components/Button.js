import React from 'react'
import PropTypes from 'prop-types'
import { View, TouchableHighlight, Image, ViewPropTypes, StyleSheet } from 'react-native'
import colors from '../utils/colors'
import { Heading3 } from '../components'

export const types = {
  FILLED: 'FILLED',
  OUTLINED: 'OUTLINED',
}

export const sizes = {
  NORMAL: 'NORMAL',
  SMALL: 'SMALL',
}

const states = {
  ENABLED: 'ENABLED',
  DISABLED: 'DISABLED',
}

const Button = (props) => {
  const { title, type, size, icon, disabled } = props
  const typeStyle = typeStyles[disabled ? states.DISABLED : states.ENABLED][type]
  const sizeStyle = sizeStyles[size]

  return (
    <TouchableHighlight
      {...props}
      style={[styles.button, typeStyle.button, sizeStyle.button, props.style]}
    >
      <View style={styles.row}>
        {icon && <Image source={icon} style={[styles.icon, typeStyle.icon]} />}
        <Heading3 keyName={title} style={[styles.text, typeStyle.text]} />
      </View>
    </TouchableHighlight>
  )
}

Object.assign(Button, types, sizes, states)

Button.propTypes = {
  style: ViewPropTypes.style,
  title: PropTypes.string.isRequired,
  type: PropTypes.oneOf(Object.keys(types)),
  icon: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  size: PropTypes.oneOf(Object.keys(sizes)),
  disabled: PropTypes.bool,
}

Button.defaultProps = {
  type: types.FILLED,
  size: sizes.NORMAL,
  disabled: false,
}

const styles = StyleSheet.create({
  row: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    height: 50,
    borderRadius: 24,
    borderWidth: 2,
    justifyContent: 'center',
    shadowColor: colors.brightTurquoise,
    shadowOpacity: 0.22,
    shadowRadius: 15,
    shadowOffset: {
      height: 10,
    },
    elevation: 10,
  },
  text: {
    textAlign: 'center',
  },
  icon: {
    width: 20,
    height: 20,
    marginRight: 9,
  },
})

const typeStyles = {
  [states.ENABLED]: {
    [types.FILLED]: StyleSheet.create({
      button: {
        backgroundColor: colors.governorBay,
        borderColor: colors.governorBay,
      },
      text: {
        color: colors.white,
      },
      icon: {
        tintColor: colors.white,
      },
    }),
    [types.OUTLINED]: StyleSheet.create({
      button: {
        backgroundColor: colors.white,
        borderColor: colors.governorBay,
      },
      text: {
        color: colors.governorBay,
      },
      icon: {
        tintColor: colors.governorBay,
      },
    }),
  },
  [states.DISABLED]: {
    [types.FILLED]: StyleSheet.create({
      button: {
        backgroundColor: colors.mischka,
        borderColor: colors.mischka,
      },
      text: {
        color: colors.white,
      },
      icon: {
        tintColor: colors.white,
      },
    }),
    [types.OUTLINED]: StyleSheet.create({
      button: {
        backgroundColor: colors.white,
        borderColor: colors.mischka,
      },
      text: {
        color: colors.mischka,
      },
      icon: {
        tintColor: colors.mischka,
      },
    }),
  },
}

const sizeStyles = {
  [sizes.NORMAL]: StyleSheet.create({
    button: {
      width: 300,
    },
  }),
  [sizes.SMALL]: StyleSheet.create({
    button: {
      width: 150,
    },
  }),
}

export default Button
