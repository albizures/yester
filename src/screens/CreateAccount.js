import React, { Component } from 'react'
import { StyleSheet, Dimensions, View } from 'react-native'
import PropTypes from 'prop-types'

import Translate from '../components/Translate'
import Container from '../components/Container'
import Button from '../components/Button'
import styles from '../styles/common'
import colors from '../utils/colors'
import icons from '../utils/icons'

export default class CreateAccount extends Component {
  static propTypes = {
    navigation: PropTypes.object.isRequired,
  }

  onPressFB = () => {
    this.props.navigation.navigate('Facebook')
  }

  onPressSignIn = () => {
    this.props.navigation.navigate('SignIn')
  }

  render () {
    return (
      <Container>
        <View style={[localStyles.container]}>
          <Translate style={[styles.h1, localStyles.margin, {textAlign: 'center'}]}
            keyName='common.upperTitle' />
          <Translate style={[styles.title, localStyles.margin, {fontWeight: 'bold', textAlign: 'center'}]}
            keyName='createAccount.begin' />

          <Button onPress={this.onPressFB}
            title={'createAccount.continue'} type={'filled'} size='normal' icon={icons.fb} />

          <Translate style={[styles.body1, localStyles.margin]}
            keyName='createAccount.recommendation' />

          <View style={localStyles.row}>
            <View style={[styles.divider, {width: 30}]} />
            <Translate style={[styles.title]} keyName='createAccount.or' />
            <View style={[styles.divider, {width: 30}]} />
          </View>

          <Button onPress={this.onPressSignIn}
            title={'createAccount.create'} type={'outlined'} size='normal' icon={0} />

        </View>
      </Container>
    )
  }
}

let { height, width } = Dimensions.get('window')
const localStyles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    alignContent: 'center',
    paddingTop: height * 0.075,
    paddingBottom: height * 0.25,
    paddingHorizontal: width * 0.08,
    backgroundColor: colors.white,
  },
  margin: {
    marginBottom: height * 0.03,
  },
  row: {
    flex: 1,
    flexDirection: 'row',
    width: width * 0.3,
    justifyContent: 'space-between',
    alignItems: 'center',
    alignSelf: 'center',
    backgroundColor: colors.white,
  },
})
