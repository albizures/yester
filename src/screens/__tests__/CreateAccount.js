import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

import CreateAccount from '../CreateAccount';
const navigation = {
	navigate: jest.fn(),
};

describe.skip('src/screens/CreateAccount.js', () => {
	it('should render', () => {
		const tree = renderer
			.create(<CreateAccount navigation={navigation} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	describe('when button to continue with FB is pressed', () => {
		it('should navigate to Facebook screen', () => {
			const shallowCreateAccount = shallow(
				<CreateAccount navigation={navigation} />,
			);
			const shallowTouchableHighlights = shallowCreateAccount.find(Button);
			const shallowButtonToFacebook = shallowTouchableHighlights.at(0);
			shallowButtonToFacebook.prop('onPress')();

			// expect(navigation.navigate).toHaveBeenCalledTimes(1)
			expect(navigation.navigate).toHaveBeenCalledWith('Facebook');
		});
	});

	describe('when button Create Account is pressed', () => {
		it('should navigate to Setuo screen', () => {
			const shallowCreateAccount = shallow(
				<CreateAccount navigation={navigation} />,
			);
			const shallowTouchableHighlights = shallowCreateAccount.find(Button);
			const shallowButtonToSignIn = shallowTouchableHighlights.at(1);

			shallowButtonToSignIn.prop('onPress')();

			// expect(navigation.navigate).toHaveBeenCalledTimes(2)
			expect(navigation.navigate).toHaveBeenCalledWith('Setup');
		});
	});
});
