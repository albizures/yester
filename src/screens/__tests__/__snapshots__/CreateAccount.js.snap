// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/screens/CreateAccount.js should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignContent": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "column",
          "paddingBottom": 333.5,
          "paddingHorizontal": 60,
          "paddingTop": 100.05,
        },
      ]
    }
  >
    <Text
      keyName="common.upperTitle"
      style={
        Array [
          Object {
            "color": "#313131",
            "fontFamily": "Karla-Regular",
            "fontSize": 50,
            "fontWeight": "bold",
            "textAlign": "left",
          },
          Object {
            "marginBottom": 40.019999999999996,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      YESTER
    </Text>
    <Text
      keyName="createAccount.begin"
      style={
        Array [
          Object {
            "color": "#313131",
            "fontFamily": "Karla-Regular",
            "fontSize": 20,
            "textAlign": "left",
          },
          Object {
            "marginBottom": 40.019999999999996,
          },
          Object {
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    >
      Start it's simple. 
Let's begin with your account.
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          undefined,
          Object {
            "borderRadius": 24,
            "borderWidth": 2,
            "height": 50,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "width": 300,
          },
          Object {
            "backgroundColor": "#4241B3",
            "borderColor": "#4241B3",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/fb-no-oficial.png",
            }
          }
          style={
            Array [
              Object {
                "height": 20,
                "width": 20,
              },
              Object {
                "tintColor": "#fff",
              },
            ]
          }
        />
        <Text
          keyName="createAccount.continue"
          style={
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "color": "#fff",
                "paddingLeft": 9,
              },
            ]
          }
        >
          Continue with Facebook
        </Text>
      </View>
    </View>
    <Text
      keyName="createAccount.recommendation"
      style={
        Array [
          Object {
            "color": "#313131",
            "fontFamily": "Karla-Regular",
            "fontSize": 16,
            "textAlign": "center",
          },
          Object {
            "marginBottom": 40.019999999999996,
          },
        ]
      }
    >
      Recommended, to speed things up.
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": 225,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#D5D3DD",
              "borderColor": "#D5D3DD",
              "borderRadius": 53.36,
              "borderWidth": 1,
              "height": 13.34,
            },
            Object {
              "width": 30,
            },
          ]
        }
      />
      <Text
        keyName="createAccount.or"
        style={
          Array [
            Object {
              "color": "#313131",
              "fontFamily": "Karla-Regular",
              "fontSize": 20,
              "textAlign": "left",
            },
          ]
        }
      >
        Or
      </Text>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#D5D3DD",
              "borderColor": "#D5D3DD",
              "borderRadius": 53.36,
              "borderWidth": 1,
              "height": 13.34,
            },
            Object {
              "width": 30,
            },
          ]
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          undefined,
          Object {
            "borderRadius": 24,
            "borderWidth": 2,
            "height": 50,
            "justifyContent": "center",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "width": 300,
          },
          Object {
            "backgroundColor": "#fff",
            "borderColor": "#4241B3",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          keyName="createAccount.create"
          style={
            Array [
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "color": "#4241B3",
              },
            ]
          }
        >
          Create account
        </Text>
      </View>
    </View>
  </View>
</View>
`;
